sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java

```diff
@@ -0,0 +1,1420 @@
+                                                                    @PathParam("key-name") String keyName,
+                                                                    @QueryParam("api-version") String apiVersion,
+                                                                    @HeaderParam("Accept") String accept,
+                                                                    Context context);
+
+        @Put("/keys/{key-name}/rotationpolicy")
+        @ExpectedResponses({200})
+        @UnexpectedResponseExceptionType(code = {404}, value = ResourceNotFoundException.class)
+        @UnexpectedResponseExceptionType(HttpResponseException.class)
+        Mono<Response<KeyRotationPolicy>> updateKeyRotationPolicyAsync(@HostParam("url") String url,
+                                                                  @PathParam("key-name") String keyName,
+                                                                  @QueryParam("api-version") String apiVersion,
+                                                                  @BodyParam("application/json") KeyRotationPolicy keyRotationPolicy,
+                                                                  @HeaderParam("Accept") String accept,
+                                                                  Context context);
+
+        @Put("/keys/{key-name}/rotationpolicy")
+        @ExpectedResponses({200})
+        @UnexpectedResponseExceptionType(code = {404}, value = ResourceNotFoundException.class)
+        @UnexpectedResponseExceptionType(HttpResponseException.class)
+        Response<KeyRotationPolicy> updateKeyRotationPolicy(@HostParam("url") String url,
+                                                                  @PathParam("key-name") String keyName,
+                                                                  @QueryParam("api-version") String apiVersion,
+                                                                  @BodyParam("application/json") KeyRotationPolicy keyRotationPolicy,
+                                                                  @HeaderParam("Accept") String accept,
+                                                                  Context context);
+    }
+
+    public Mono<Response<KeyVaultKey>> createKeyWithResponseAsync(String name, KeyType keyType, Context context) {
+        KeyRequestParameters parameters = new KeyRequestParameters().setKty(keyType);
+        return service.createKeyAsync(vaultUrl, name, keyServiceVersion.getVersion(), ACCEPT_LANGUAGE, parameters,
+                CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+            .doOnRequest(ignored -> logger.verbose("Creating key - {}", name))
+            .doOnSuccess(response -> logger.verbose("Created key - {}", response.getValue().getName()))
+            .doOnError(error -> logger.warning("Failed to create key - {}", name, error));
```

Comment:
Do we want to remove this logging or replicate the logic in the synchronous call paths?

Reply:
I think we should keep it for async and remove for sync. The sync stack trace is going to have enough information.